api/client/build.go

```diff
@@ -230,13 +247,16 @@ func (cli *DockerCli) CmdBuild(args ...string) error {
 		return err
 	}
 
-	err = jsonmessage.DisplayJSONMessagesStream(response.Body, cli.out, cli.outFd, cli.isTerminalOut)
+	err = jsonmessage.DisplayJSONMessagesStream(response.Body, buildBuff, cli.outFd, cli.isTerminalOut)
 	if err != nil {
 		if jerr, ok := err.(*jsonmessage.JSONError); ok {
 			// If no error code is set, default to 1
 			if jerr.Code == 0 {
 				jerr.Code = 1
 			}
+			if *suppressOutput {
+				fmt.Fprintln(cli.err, progBuff, buildBuff)
```

Comment:
let's replace this with:
```
fmt.Fprintf(cli.err, "%s%s", progBuff, buildBuff)
```
otherwise we get an extra newline between when -q is used and when its not. In particular see these two:
```
docker build -q - < tmp/Dockerfile
Sending build context to Docker daemon 2.048 kB
Step 1 : FROM ubuntu
 ---> 4b1e42b414f6
Step 2 : RUN pwd1
 ---> Running in 8da0f61403cd
/bin/sh: 1: pwd1: not found

The command '/bin/sh -c pwd1' returned a non-zero code: 127
root@docker:~/docker# docker build - < tmp/Dockerfile
Sending build context to Docker daemon 2.048 kB
Step 1 : FROM ubuntu
 ---> 4b1e42b414f6
Step 2 : RUN pwd1
 ---> Running in d7893abb4e7a
/bin/sh: 1: pwd1: not found
The command '/bin/sh -c pwd1' returned a non-zero code: 127
```

Notice the extra newline before the final error message in the first case.  2nd case is with the new print.